python/lsst/meas/algorithms/stamps.py

```diff
@@ -0,0 +1,335 @@
+        A factory function defined on a dataclass for constructing
+        stamp objects a la `lsst.meas.alrogithm.Stamp`
+    options : `PropertySet`
+        A collection of parameters.  If certain keys are available
+        (``llcX``, ``llcY``, ``width``, ``height``), a bounding box
+        is constructed and passed to the ``FitsReader`` in order
+        to return a sub-image.
+
+    Returns
+    -------
+    stamps, metadata : `list` of `lsst.afw.image.MaskedImageF`, PropertySet
+        A tuple of a list of masked images and a collection of metadata.
+    """
+    # extract necessary info from metadata
+    metadata = afwFits.readMetadata(filename, hdu=0)
+    nStamps = metadata["N_STAMPS"]
+    # check if a bbox was provided
+    kwargs = {}
+    if options and options.exists("llcX"):
+        llcX = options["llcX"]
+        llcY = options["llcY"]
+        width = options["width"]
+        height = options["height"]
+        bbox = Box2I(Point2I(llcX, llcY), Extent2I(width, height))
+        kwargs["bbox"] = bbox
+    # read stamps themselves
+    stamps = []
+    for idx in range(nStamps):
+        n_parts = 1  # Always have at least an image
+        for key in ['HAS_MASK', 'HAS_VARIANCE']:
+            if metadata[key]:
+                n_parts += 1
+        parts = {}
+        imReader = afwImage.ImageFitsReader(filename, hdu=n_parts*idx + 1)
+        parts['image'] = imReader.read(**kwargs)
+        if metadata['HAS_MASK']:
+            # Alwasy stored right after image if present
+            maskReader = afwImage.MaskFitsReader(filename, hdu=n_parts*idx + 2)
+            parts['mask'] = maskReader.read(**kwargs)
+        if metadata['HAS_VARIANCE']:
+            # Maybe either right after image or two after
+            varReader = afwImage.ImageFitsReader(filename, hdu=n_parts*idx + n_parts)
+            parts['variance'] = varReader.read(**kwargs)
+        maskedImage = afwImage.MaskedImageF(**parts)
+        stamps.append(stamp_factory(maskedImage, metadata.toDict(), idx))
+    return stamps, metadata
+
+
+@dataclass
+class Stamp:
+    """Single stamp
+
+    Parameters
+    ----------
+    stamp : `lsst.afw.image.MaskedImageF`
+        The actual pixel values for the postage stamp
+    ra : `lsst.geom.Angle`
+        The Right Ascention of the center of the stamp
+    dec : `lsst.geom.Angle`
+        The Declination of the center of the stamp
+    size : `int`
+        The size of the stamp in pixels
+    """
+    stamp: afwImage.maskedImage.MaskedImageF
+    ra: Angle
+    dec: Angle
+    size: int
+
+    @classmethod
+    def factory(cls, stamp, metadata, index):
+        """A factory method to construct an instance of this class
+        given a masked image and some metadata.
+
+        Parameteres
+        -----------
+        stamp : `lsst.afw.image.MaskedImageF`
+            The pixel data to associate with the instance
+        metadata : `dict`
+            A mapping of metadata used to populate this instance
```

Comment:
It's a dict that has three required keys and whose values must be a sequence.

Reply:
I think I've addressed this, but let me know if it's not complete enough